### PR TITLE
Fix RegularSequence.subsequence for zero sequence

### LINKED ISSUE #37282: RegularSequence.subsequence broken for zero sequence
### Steps To Reproduce

```python
R = RegularSequenceRing(2, QQ)
z = R.zero()
z.subsequence(1, 1)
```

### Expected Behavior

```
2-regular sequence 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...
```

### Actual Behavior

```
Traceback (most recent call last):
...
  File "/local/cheuberg/local/sage-10.3.beta7/src/sage/combinat/regular_sequence.py", line 850, in subsequence
    vector(chain.from_iterable(
  File "sage/modules/free_module_element.pyx", line 587, in sage.modules.free_module_element.vector
  File "sage/modules/free_module_element.pyx", line 690, in sage.modules.free_module_element.prepare
TypeError: unable to find a common ring for all elements
```

### Additional Information

Cc: @dkrenn 

### Environment

```markdown
- **OS**: Ubuntu 20.04
- **Sage Version**: 10.3.beta7
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by cheuberg**
Fixes #37282 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

Cc: @dkrenn

### CODE REVIEW

File: src/sage/combinat/regular_sequence.py [hunk 1]
```diff
@@ -762,6 +762,12 @@ def subsequence(self, a, b):
 
             sage: S.regenerated().subsequence(1, -4)
             2-regular sequence 0, 0, 0, 0, 1, 3, 6, 9, 12, 18, ...
+
+        Check that the zero sequence is handeled correctly (issue:`37282`)
```
Comment (S17A05) [inline src/sage/combinat/regular_sequence.py:766]: Juist

Comment (S17A05) [inline src/sage/combinat/regular_sequence.py]: It should be "handled" instead of "handeled".

**Review by S17A05 — COMMENTED** [review]
Just a small typo, otherwise LGTM.

File: src/sage/combinat/regular_sequence.py (hunk 1 above)
Comment (cheuberg) [reply src/sage/combinat/regular_sequence.py]: Thank you, fixed in 6b3480e.

**Review by S17A05 — APPROVED** [review]
Thanks for the fix - LGTM.

**Pull request merged**